lib/datadog/core/configuration/option.rb

```diff
@@ -13,11 +13,11 @@ module Precedence
           # Remote configuration provided through the Datadog app.
           REMOTE_CONFIGURATION = [2, :remote_configuration].freeze
 
-          # Configuration provided in Ruby code, in this same process.
+          # Configuration provided in Ruby code, in this same process
+          # or via Environment variable
           PROGRAMMATIC = [1, :programmatic].freeze
```

Comment:
`PROGRAMMATIC` feels very weird. A couple suggestions:

- it's configuration as code
  
  ```
  CODE = :code
  ```

- configuration is written by using a DSL:
  
  ```
  DSL = :dsl
  ```

- but the above are quite "technical" so to speak, so I quite like this one:
  
  ```
  LOCAL = :local
  ```

Why do I like it?:

- `local` is squarely set within the app
- `remote` is telecontrolled from some far away place
- `env` is from outside the app (literally the definition of "environment"), and possibly globally set at the infrastructure level
- if there is someday such a thing as configuring from, say, a YAML file, `file` is from outside the app as well

So overall I feel they are all quite short and clear when contextualised.